src/frontend/packages/lib_video/src/components/common/VideoWidgetProvider/widgets/DeleteVideo/index.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  ConfirmationModal,
+  FoldableItem,
+  report,
+  useCurrentResourceContext,
+} from '@lib-video/../../lib_components/src';
+import { Button } from 'grommet';
+import { useState } from 'react';
+import toast from 'react-hot-toast';
+import { defineMessages, useIntl } from 'react-intl';
+import { useHistory } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { useDeleteVideo } from '@lib-video/api/useDeleteVideo';
+import { useCurrentVideo } from '@lib-video/hooks';
+
+const messages = defineMessages({
+  info: {
+    defaultMessage:
+      'DANGER ZONE : This widget allows you to delete the VOD permanently.',
+    description: 'Info of the widget used for deleting the video.',
+    id: 'components.DeleteVideo.info',
+  },
+  title: {
+    defaultMessage: 'DANGER ZONE',
+    description: 'Title of the widget used for deleting the video.',
+    id: 'components.DeleteVideo.title',
+  },
+  confirmDeleteTitle: {
+    defaultMessage: 'Confirm delete video',
+    description: 'Title of the widget used for confirmation.',
+    id: 'components.DeleteVideoConfirm.title',
+  },
+  confirmDeleteText: {
+    defaultMessage:
+      'Are you sure you want to delete this VOD ? This action is irreversible.',
+    description: 'Text of the widget used for confirmation.',
+    id: 'components.DeleteVideoConfirm.text',
+  },
+  deleteButtonText: {
+    defaultMessage: 'Delete video',
+    description: 'Text of the delete button.',
+    id: 'components.DeleteVideoButton.text',
+  },
+  videoDeleteSuccess: {
+    defaultMessage: 'Video successfully deleted',
+    description: 'Text of the delete confirmation toast.',
+    id: 'components.deleteVideoSuccess.text',
+  },
+  videoDeleteError: {
+    defaultMessage: 'Failed to delete the video',
+    description: 'ext of the delete error toast..',
+    id: 'components.videoDeleteError.text',
+  },
+});
+
+const StyledAnchorButton = styled(Button)`
+  height: 50px;
+  font-family: 'Roboto-Medium';
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+export const DeleteVideo = () => {
+  const intl = useIntl();
+  const [context] = useCurrentResourceContext();
+  const history = useHistory();
+  const [showDeleteConfirmationModal, setShowDeleteConfirmationModal] =
+    useState(false);
+  const video = useCurrentVideo();
+  const deleteVideo = useDeleteVideo({
+    onSuccess: () => {
+      toast.success(intl.formatMessage(messages.videoDeleteSuccess), {
+        position: 'bottom-center',
+      });
+      setShowDeleteConfirmationModal(false);
+      history.push(context.resourcePath || `/my-contents/videos`);
```

Comment:
I think it's a bad idea to have this path harcoded. We can maybe use `history.goBack()` ?

```
if (context.resourcePath) {
    history.push(context.resourcePath);
} else {
    history.goBack();
}
```

WDYT ?

Reply:
I think so too, I just didn't know I could do it like this, thanks